linked-list.js

```diff
@@ -0,0 +1,399 @@
+      const newNode = { data: currentListNode.data, next: null };
+
+      if (!list.head) {
+        list.head = newNode;
+        list.tail = newNode;
+      } else {
+        list.tail.next = newNode;
+        list.tail = newNode;
+      }
+
+      currentListNode = currentListNode.next;
+    }
+
+    return list;
+  }
+  return { head: null, tail: null };
+}
+
+/**
+ * adds data into a fresh node towards the end of the list
+ */
+
+export function addItemToList(listRef, data) {
+  // assert(listRef !== undefined && listRef !== null);
+  const newNode = { data, next: null };
+  if (listRef.head === null) {
+    listRef.head = newNode;
+  } else {
+    let lastNode = listRef.head;
+    while (lastNode.next !== null) {
+      lastNode = lastNode.next;
+    }
+    lastNode.next = newNode;
+    listRef.tail = newNode;
+  }
+
+  return newNode;
+}
+export function arrayFromList(listRef) {
+  const array = [];
+  let lastNode = listRef.head;
+  while (lastNode.next !== null) {
+    array.push(lastNode.data);
+    lastNode = lastNode.next;
+  }
+  array.push(lastNode.data);
+  return array;
+}
+
+export function removeFromEnd(listRef) {
+  if (listRef.head === null) {
+    return;
+  }
+  if (listRef.head.next === null) {
+    listRef.head = null;
+    listRef.tail = null;
+    return;
+  }
+  let prevNode = null;
+  let lastNode = listRef.head;
+  while (lastNode.next !== null) {
+    prevNode = lastNode;
+    lastNode = lastNode.next;
+  }
+  prevNode.next = null;
+  listRef.tail = prevNode;
+}
+
+export function insertAfter(listRef, listNode, data) {
+  if (listRef.head === null || listNode === null) {
+    return null;
+  }
+
+  const newNode = { data, next: null };
+
+  let lastNode = listRef.head;
+  while (lastNode !== null) {
+    if (lastNode === listNode) {
+      newNode.next = lastNode.next;
+      lastNode.next = newNode;
+      return newNode;
+    }
+    lastNode = lastNode.next;
+  }
+
+  return null;
+}
+
+export function insertBefore(listRef, listNode, data) {
+  if (listRef.head === null || listNode === null) {
+    return null;
+  }
+  const newNode = { data, next: null };
+  let lastNode = listRef.head;
+  while (lastNode !== null) {
+    if (lastNode.next === listNode) {
+      newNode.next = lastNode.next;
+      lastNode.next = newNode;
+      return newNode;
+    }
+    lastNode = lastNode.next;
+  }
+  return null;
+}
+
+// eslint-disable-next-line consistent-return
+export function removeItem(listRef, data) {
+  if (listRef.head === null || data === null) {
+    return null;
+  }
+  if (listRef.head.next === null) {
+    listRef.head = null;
+    // eslint-disable-next-line no-param-reassign
+    listRef.tail = null;
+    // eslint-disable-next-line consistent-return
+    return;
+  }
+  let prevNode = null;
+  let lastNode = listRef.head;
+  while (lastNode !== null) {
+    if (lastNode.data === data) {
+      prevNode.next = lastNode.next;
+      // eslint-disable-next-line consistent-return

```

Comment:
We need to stop disabling eslint like this.  Guess such disabling only defeats the purpose of ESLint.

Reply:
Implemented the changes and modified the code by not disabling it.